includes/abstracts/abstract-wc-data.php

```diff
@@ -364,11 +364,9 @@ public function add_meta_data( $key, $value, $unique = false ) {
 	public function update_meta_data( $key, $value, $meta_id = '' ) {
 		$this->maybe_read_meta_data();
 		if ( $array_key = $meta_id ? array_keys( wp_list_pluck( $this->meta_data, 'id' ), $meta_id ) : '' ) {
-			$this->meta_data[ current( $array_key ) ] = (object) array(
-				'id'    => $meta_id,
-				'key'   => $key,
-				'value' => $value,
-			);
+			$meta = $this->meta_data[ current( $array_key ) ];
```

Comment:
No, if that block is evaluated TRUE means that you already provided the `meta_id` value and you found a meta data object by their ID. `update_meta_data( 'foobar', 'some value', 5)` is quite specific, update the `foobar` metadata with the ID=5. If `foobar` is found its ID will already by 5.